File: examples/producer/typescript/src/producer/beacons/beacons.likemessage.ts
import {
	Context,
	Producer,
	Identification,
	Filter,
	Protocol,
} from "../implementation/beacons";
import { Scope } from "../implementation/scope";

export function emit(
	beacon: Protocol.Beacons.LikeMessage,
	scope: Scope
): Promise<void> {
	return Promise.reject(
		new Error(`Handler for Protocol.Beacons.LikeMessage isn't implemented.`)
	);
}

File: examples/producer/rust/src/producer/beacons/beacons_likemessage.rs
use super::{identification, producer::Control, protocol, Context, scope::Scope};
use clibri::server;

#[allow(unused_variables)]
pub async fn emit<E: server::Error, C: server::Control<E>>(
    beacon: &protocol::Beacons::LikeMessage,
    scope: &mut Scope<'_, E, C>,
) -> Result<(), String> {
    println!("Handler for protocol::Beacons::LikeMessage isn't implemented");
    Ok(())
}